### PR TITLE
wait for provision to complete before install test agent

### PR DESCRIPTION
**Pull request by nagworld9**
## Description

wait for provision to complete before install test agent.

Issue # 

---

### PR information
- [ ] The title of the PR is clear and informative.
- [ ] There are a small number of commits, each of which has an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).
- [ ] If applicable, the PR references the bug/issue that it fixes in the description.
- [ ] New Unit tests were added for the changes made

### Quality of Code and Contribution Guidelines
- [ ] I have read the [contribution guidelines](https://github.com/Azure/WALinuxAgent/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Review by maddieford — DISMISSED** [review]

File: tests_e2e/orchestrator/scripts/install-agent [hunk 1]
```diff
@@ -85,19 +85,19 @@ echo "Agent Version:"
 $python "$waagent" --version
 echo "Service Status:"
 
-# Sometimes the service can take a while to start; give it a few minutes,
+# Sometimes the service can take a while to start and provision completed; give it a few minutes,
```
Comment (narrieta) [inline tests_e2e/orchestrator/scripts/install-agent:88]: I'd update the comment to describe that we need to wait for the provisioning code before stopping the agent's service to do the test setup

File: tests_e2e/orchestrator/scripts/install-agent [hunk 2]
```diff
@@ -85,19 +85,19 @@ echo "Agent Version:"
...
 started=false
 for i in {1..6}
 do
-  if service-status $service_name; then
+  if [[ -f /var/lib/waagent/provisioned ]]; then
     started=true
     break
   fi
-  echo "Waiting for service to start..."
+  echo "Waiting for service to complete provision..."
```
Comment (narrieta) [inline tests_e2e/orchestrator/scripts/install-agent:96]: "Waiting for agent to complete provisioning."?

File: tests_e2e/orchestrator/scripts/install-agent [hunk 3]
```diff
@@ -85,19 +85,19 @@ echo "Agent Version:"
...
   sleep 30
 done
 if [ $started == false ]; then
-  echo "Service failed to start."
+  echo "Service failed to complete provision."
```
Comment (narrieta) [inline tests_e2e/orchestrator/scripts/install-agent:100]: "Provisioning did not completed in the given timeout (cannot find /var/lib/waagent/provisioned)"?

File: tests_e2e/orchestrator/scripts/install-agent (hunk 1 above)
Comment (nagworld9) [reply tests_e2e/orchestrator/scripts/install-agent]: addressed

**Review by narrieta — APPROVED** [review]

**Review by maddieford — APPROVED** [review]

**Pull request merged**